src/vm/virtualMachine.ts

```diff
@@ -23,16 +23,39 @@ import {
   ResLogic,
 } from './instruction';
 
+type TraceEntry = {
+  pc: Relocatable;
+  ap: Relocatable;
+  fp: Relocatable;
+};
+
+type RelocatedTraceEntry = {
+  pc: Felt;
+  ap: Felt;
+  fp: Felt;
+};
+
+type RelocatedMemory = {
+  address: number;
+  value: Felt;
+};
+
 export class VirtualMachine {
   private currentStep: bigint;
   memory: Memory;
   pc: Relocatable;
   ap: Relocatable;
   fp: Relocatable;
+  trace: TraceEntry[];
+  relocatedTrace: RelocatedTraceEntry[];
+  relocatedMemory: RelocatedMemory[];
 
   constructor() {
     this.currentStep = 0n;
     this.memory = new Memory();
+    this.relocatedMemory = [];
+    this.trace = [];
+    this.relocatedTrace = [];
```

Comment:
<img width="279" alt="image" src="https://github.com/kkrt-labs/cairo-vm-ts/assets/18620296/72a5b068-0ce3-4a01-af10-b9d627a47598">
plz decide an order between memory/trace/relocatedM/relocatedTrace and memory/relocatedM/trace/relocatedT and keep